test/export/InstanceExporter.test.ts

```diff
@@ -897,6 +901,146 @@ describe('InstanceExporter', () => {
+      const cardRule = new CardRule('contact.gender');
+      cardRule.min = 1;
+      cardRule.max = '1';
+      patient.rules.push(cardRule);
+      const fixedValueRule1 = new FixedValueRule('contact[0].relationship');
+      fixedValueRule1.fixedValue = new FshCode('Looking for love');
+      patientInstance.rules.push(fixedValueRule1);
+      const fixedValueRule2 = new FixedValueRule('contact[1].relationship');
+      fixedValueRule2.fixedValue = new FshCode('Complicated');
+      patientInstance.rules.push(fixedValueRule2);
+      exportInstance(patientInstance);
+      const messages = loggerSpy.getAllMessages('error');
+      expect(messages[messages.length - 2]).toMatch(
+        /Patient.contact.gender.*File: PatientInstance\.fsh.*Line: 10 - 20/s
+      );
+      expect(messages[messages.length - 1]).toMatch(
+        /Patient.contact.gender.*File: PatientInstance\.fsh.*Line: 10 - 20/s
+      );
+    });
+
+    it('should log an error when an [x] element is not present', () => {
+      const cardRule = new CardRule('deceased[x]');
+      cardRule.min = 1;
+      cardRule.max = '1';
+      patient.rules.push(cardRule);
+      exportInstance(patientInstance);
+      const messages = loggerSpy.getAllMessages('error');
+      expect(messages[messages.length - 1]).toMatch(
+        /Patient.deceased\[x\].*File: PatientInstance\.fsh.*Line: 10 - 20/s
+      );
+    });
+
+    it('should not log an error when an [x] element is present', () => {
+      const originalLength = loggerSpy.getAllMessages('error').length;
+      const cardRule = new CardRule('deceased[x]');
+      cardRule.min = 1;
+      cardRule.max = '1';
+      patient.rules.push(cardRule);
+      const fixedValueRule = new FixedValueRule('deceasedBoolean');
+      fixedValueRule.fixedValue = true;
+      patientInstance.rules.push(fixedValueRule);
+      exportInstance(patientInstance);
+      expect(loggerSpy.getAllMessages('error').length).toBe(originalLength);
+    });
+
+    it('should log an error when a required sliced element is not present', () => {
+      const fixedValueRule = new FixedValueRule('result[Cholesterol]');
+      fixedValueRule.fixedValue = new FshReference('Fsh are friends');
+      lipidInstance.rules.push(fixedValueRule);
+      exportInstance(lipidInstance);
+      const messages = loggerSpy.getAllMessages('error');
+      expect(messages[messages.length - 4]).toMatch(
+        /DiagnosticReport.status.*File: LipidInstance\.fsh.*Line: 10 - 20/s
+      );
+      expect(messages[messages.length - 3]).toMatch(
+        /DiagnosticReport.result.*File: LipidInstance\.fsh.*Line: 10 - 20/s
+      );
+      expect(messages[messages.length - 2]).toMatch(
+        /DiagnosticReport.result:Triglyceride.*File: LipidInstance\.fsh.*Line: 10 - 20/s
+      );
+      expect(messages[messages.length - 1]).toMatch(
+        /DiagnosticReport.result:HDLCholesterol.*File: LipidInstance\.fsh.*Line: 10 - 20/s
+      );
+    });
```

Comment:
Can we add one test that ensures that it validates cardinalities that came from a parent SD not authored in FSH?  I think all of these tests are based on constraints in the direct profile -- but it would be good to have a test that proves it works when the constraint is inherited too.  `Observation.code` is `1..1` -- so that would be a pretty easy one.

Maybe two tests actually -- one that tests that an `InstanceOf: Observation` requires `code` -- and another that tests that an `InstanceOf: MyObservationProfile` requires `code`.  Then I think we have all our bases covered.